src/components/Button/BaseButton.tsx

```diff
@@ -130,30 +195,67 @@ export const BaseButton: FC<InternalButtonProps> = React.forwardRef(
-                {iconExists && textExists && (
-                    <span>
-                        {getButtonIcon(icon)}
-                        {getButtonText(buttonTextClassNames, text)}
-                    </span>
+            <>
+                <button
+                    {...rest}
+                    ref={ref}
+                    aria-checked={toggle ? !!checked : undefined}
+                    aria-disabled={allowDisabledFocus}
+                    aria-label={ariaLabel}
+                    aria-pressed={toggle ? !!checked : undefined}
+                    defaultChecked={checked}
+                    disabled={disabled}
+                    className={buttonBaseClassNames}
+                    id={id}
+                    onClick={!allowDisabledFocus ? onClick : null}
+                    style={style}
+                    type={htmlType}
+                >
+                    {iconExists && !textExists && getButtonIcon(icon)}
+                    {iconExists && textExists && (
+                        <span>
+                            {getButtonIcon(icon)}
+                            {getButtonText(buttonTextClassNames, text)}
+                        </span>
+                    )}
+                    {!iconExists && getButtonText(buttonTextClassNames, text)}
+                </button>
+                {split && (
+                    <>
+                        <button
+                            aria-checked={
+                                split ? !!splitButtonChecked : undefined
+                            }
+                            aria-disabled={splitButtonProps?.allowDisabledFocus}
+                            aria-label={splitButtonProps?.ariaLabel}
+                            aria-pressed={
+                                split ? !!splitButtonChecked : undefined
+                            }
+                            defaultChecked={splitButtonChecked}
+                            disabled={splitButtonProps?.disabled}
+                            className={splitButtonClassNames}
+                            id={splitButtonProps?.id}
+                            onClick={
+                                !splitButtonProps?.allowDisabledFocus
+                                    ? onContextMenu
+                                    : null
+                            }
+                            style={splitButtonProps?.style}
+                            type="button"
+                        >
+                            {getButtonIcon(
+                                splitButtonProps?.icon || splitButtonChecked
+                                    ? IconName.mdiChevronDown
+                                    : IconName.mdiChevronUp
+                            )}
+                            <span
+                                className={splitDividerClassNames}
+                                aria-hidden="true"
+                            ></span>
+                        </button>
```

Comment:
Lets move this into its own component and use it here, this file is getting comlicated.